File: python/luhn/luhn.py
class Luhn:
    def __init__(self, card_num):
        self.card_num = card_num

    def valid(self):
        # Remove spaces and check if the string is empty or contains non-digit characters
        card_num = self.card_num.replace(" ", "")
        if not card_num.isdigit() or len(card_num) <= 1:
            return False

        # Convert the card number to a list of integers in reverse order
        digits = [int(digit) for digit in card_num[::-1]]

        # Double every second digit, starting from the right
        for i in range(1, len(digits), 2):
            digits[i] *= 2
            if digits[i] > 9:
                digits[i] -= 9

        # Sum all the digits
        total = sum(digits)

        # Check if the total is divisible by 10
        return total % 10 == 0

File: javascript/luhn/luhn.js
//
// This is only a SKELETON file for the 'Luhn' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const valid = (str) => {
  str = str.replace(/ /g, "");
  if(str.length < 2){
    return false;
  }

  let digits = [...str].map(c => parseInt(c));
  for(let i = digits.length - 2; i >= 0; i -= 2){
   digits[i] = (digits[i] * 2 > 9) ? digits[i] * 2 - 9 : digits[i] * 2;
  }
  return digits.reduce((sum, digit) => sum + digit) % 10 === 0;
};